client/src/featureform/register.py

```diff
@@ -1590,7 +1591,89 @@ class Customer:
             properties=properties,
         )
 
+### Problems: the user is forced to specify th entity column. If they put the same entity name as the class that they are defining the multiple features in, will it error? 
+### If the mention a different class name, then will it get applied and work?
+class MultiFeatureColumnResource(ColumnResource):
+    def __init__(
+        self,
+        dataset: SourceVariant,
+        df: pd.DataFrame,
+        entity_column: Union[Entity, str] = "",
+        variant: str = "",
+        owner: str = "",
+        inference_store: Union[str, OnlineProvider, FileStoreProvider] = "",
+        timestamp_column: str = "",
+        include_columns: List[str] = [],
+        exclude_columns: List[str] = [],
+        description: str = "",
+        schedule: str = "",
+        tags: List[str] = [],
+        properties: Dict[str, str] = {},
+    ):
+        """
+        Feature registration object.
+
+        **Example**
+        ```
+        @ff.entity
+        class Customer:
+        # Register mulitple columns from a dataframe as a feature
+            transaction_amount = ff.MultiFeature(
+                dataframe_name,
+                variant="quickstart",
+                inference_store=redis,
+                entity_column="CustomerID",
+                include_columns=["Amount", "Transaction Time"],
+            )
+        ```
 
+        Args:
+            inference_store (Union[str, OnlineProvider, FileStoreProvider]): Where to store for online serving.
+        """
+        pd_to_ff_datatype = {numpy.dtype('float64'): ScalarType.FLOAT64,
+        numpy.dtype('float32'): ScalarType.FLOAT32,
+        numpy.dtype('int64'): ScalarType.INT64,
+        numpy.dtype('int'): ScalarType.INT,
+        numpy.dtype('int32'): ScalarType.INT32,
+        numpy.dtype('O'): ScalarType.STRING,
+        numpy.dtype('str'): ScalarType.STRING,
+        numpy.dtype('bool'): ScalarType.BOOL}
+
+        register_columns = self.get_feature_columns(df, include_columns, exclude_columns, entity_column, timestamp_column)
+        register_features_list = []
+        for column_name in register_columns:
+            if timestamp_column != "":
+                feature = FeatureColumnResource(dataset[[entity_column, column_name, timestamp_column]], variant=variant, type=pd_to_ff_datatype[df[self.add_quotes(column_name)].dtype], inference_store=inference_store)
+            else:
+                feature = FeatureColumnResource(dataset[[entity_column, column_name]], variant=variant, type=pd_to_ff_datatype[df[self.add_quotes(column_name)].dtype], inference_store=inference_store)
```

Comment:
did we try it out with other providers? I am wondering if the quotes is added to Postgres only or all other providers as well?

Reply:
This should 1000% be tested on spark and not postgres